### PR TITLE
[Merged by Bors] - chore: remove more easy autoImplicit

### PR DESCRIPTION
**Pull request by grunweg**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/EditDistance/Estimator.lean [hunk 1]
```diff
@@ -23,10 +23,8 @@ This is then used in the implementation of `rewrite_search`
 to avoid needing the entire edit distance calculation in unlikely search paths.
 -/
 
-set_option autoImplicit true
-
 variable {α β δ : Type} [CanonicallyLinearOrderedAddCommMonoid δ]
```
Comment (fpvandoorn) [inline Mathlib/Data/List/EditDistance/Estimator.lean:26]: I know you didn't edit this, but this missing `*` is intentional?

**Review by fpvandoorn — COMMENTED** [review]
These changes make it clear to me how "evil" `autoImplicit` is. with it, it's hard to even see that some lemmas are only for `Type`, not `Type*`

**Comment by fpvandoorn** [comment]
bors d+

File: Mathlib/Data/List/EditDistance/Estimator.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Data/List/EditDistance/Estimator.lean]: Good question! I checked; \alpha can be generalised, but making \beta or \delta take Type* errors all over the place. I didn't think much if that can be fixed, but will leave that to a follow-up.

**Comment by grunweg** [comment]
Thanks for the quick review!
bors r+

**Pull request closed without merging**